From a71b6e2f189e90b272094b8b9cd88752ff96e785 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz93@gmail.com>
Date: Wed, 13 Mar 2024 00:34:22 -0400
Subject: [PATCH] Fix regression causing executables to not depend on libmpcdec

In revision r491 a "Gentoo patch" was merged, entitled

"Fixup installation of shared mpcdec library."

It inherently clashed with revision r467, which *also* renamed the
target in question.
--- a/mpc2sv8/CMakeLists.txt
+++ b/mpc2sv8/CMakeLists.txt
@@ -13,7 +13,7 @@ endif(MSVC)
 add_executable(mpc2sv8 mpc2sv8.c)
 
 if(SHARED)
-  target_link_libraries(mpc2sv8 mpcdec)
+  target_link_libraries(mpc2sv8 mpcdec_shared)
 else(SHARED)
   target_link_libraries(mpc2sv8 mpcdec_static)
 endif(SHARED)
--- a/mpcchap/CMakeLists.txt
+++ b/mpcchap/CMakeLists.txt
@@ -15,7 +15,7 @@ link_directories(${libmpc_BINARY_DIR}/libmpcenc)
 add_executable(mpcchap mpcchap.c dictionary.c iniparser.c ${libmpc_SOURCE_DIR}/common/tags.c)
 
 if(SHARED)
-  target_link_libraries(mpcchap mpcdec)
+  target_link_libraries(mpcchap mpcdec_shared)
 else(SHARED)
   target_link_libraries(mpcchap mpcdec_static)
 endif(SHARED)
--- a/mpccut/CMakeLists.txt
+++ b/mpccut/CMakeLists.txt
@@ -13,7 +13,7 @@ endif(MSVC)
 add_executable(mpccut mpccut.c)
 
 if(SHARED)
-  target_link_libraries(mpccut mpcdec)
+  target_link_libraries(mpccut mpcdec_shared)
 else(SHARED)
   target_link_libraries(mpccut mpcdec_static)
 endif(SHARED)
--- a/mpcdec/CMakeLists.txt
+++ b/mpcdec/CMakeLists.txt
@@ -15,7 +15,7 @@ add_executable(mpcdec_cmd mpcdec.c)
 target_link_libraries(mpcdec_cmd wavformat_static)
 
 if(SHARED)
-  target_link_libraries(mpcdec_cmd mpcdec)
+  target_link_libraries(mpcdec_cmd mpcdec_shared)
 else(SHARED)
   target_link_libraries(mpcdec_cmd mpcdec_static)
 endif(SHARED)
--- a/mpcgain/CMakeLists.txt
+++ b/mpcgain/CMakeLists.txt
@@ -13,14 +13,14 @@ FIND_LIBRARY(REPLAY_GAIN_LIBRARY NAMES replaygain replaygain_static PATHS
 include_directories(${libmpc_SOURCE_DIR}/include)
 include_directories(${REPLAY_GAIN_INCLUDE_DIR})
 
-link_directories(${libmpc_BINARY_DIR}/libmpcdec)
+link_directories(${libmpc_BINARY_DIR}/libmpcdec_shared)
 
 add_executable(mpcgain mpcgain.c)
 
 target_link_libraries(mpcgain ${REPLAY_GAIN_LIBRARY})
 
 if(SHARED)
-  target_link_libraries(mpcgain mpcdec)
+  target_link_libraries(mpcgain mpcdec_shared)
 else(SHARED)
   target_link_libraries(mpcgain mpcdec_static)
 endif(SHARED)
-- 
2.43.2

